lib/openhab/dsl/debouncer.rb

```diff
@@ -0,0 +1,241 @@
+# frozen_string_literal: true
+
+module OpenHAB
+  module DSL
+    #
+    # Provides the feature for debouncing calls to a given block.
+    # It is used by {DSL::Rules::BuilderDSL#debounce debounce rule guard} and
+    # {OpenHAB::DSL.debounce #debounce DSL method}
+    #
+    # The debouncer can filter events and only allow the events on the leading or trailing edge
+    # of the given interval. Its behavior can be customized through settings passed to its
+    # {initialize constructor}.
+    #
+    # The following timing diagram illustrates the incoming triggers and the actual executions
+    # that are allowed by {debounce} filter.
+    #
+    # - Each character position represents an instant of time,
+    #   and the space inbetween is the unit of time
+    # - The `T` line shows individual trigger events represented by numbers/letters, and
+    # - The `E` line shows the actual executions of the corresponding trigger that are
+    #   allowed by the debounce guard.
+    #
+    # ```java
+    # Leading edge trigger, interval: 5
+    # T: A.B...C...D..EFG.HI......JKLMNOPQRST....U.....
+    # E: A.....C......E....I......J....O....T....U.....
+    #
+    # Trailing edge trigger, interval: 5
+    # T: A.B...C...D..EFG.HI......JKLMNOPQRST....U.....
+    # E: |....B|....D.|....I......|....O....T....U.....
+    # Note: "|" indicates the start of the debounce period
+    # ```
```

Comment:
these charts are super helpful. can you please provide charts for `interval: 5..10`, `min_idle`, and max_interval/min_idle in combination? cause I still don't understand what those options do. It would also be nice if you provide example use cases for each combination of params (i.e. leading edge is good for doorbell throttle; normal debounce is good re-calculating based off a rapidly updating sensor, but not missing the last value, etc.)